openhtf/core/__init__.py

```diff
@@ -114,6 +114,10 @@ def _thread_proc(self):
       # Any access to self._exit_stack must be done while holding this lock.
       with self._lock:
         self._exit_stack = exit_stack
+        # We need to exit the PhaseExecutor before we tear down the plugs
+        # that a phase may be executing, so we add them in reverse order.
+        exit_stack.callback(self.test_state.plug_manager.tear_down_plugs)
+        exit_stack.callback(executor.stop)
```

Comment:
Well, if `initialize_plugs()` is never called, `tear_down_plugs()` does nothing, right? The problem I'm getting is if we're in `test_start` and attempt to exit, nothing here gets torn down.